app/js/controllers.js

```diff
@@ -0,0 +1,139 @@
+floripaApp.controller("GMapController", ['$scope', 'uiGmapGoogleMapApi', '$location', 'searchService', function ($scope, uiGmapGoogleMapApi, $location, searchService) {
+
+    $scope.goBack = function () {
+        $location.url('/');
+    };
+
+    $scope.search = function () {
+        searchService.streetName = $scope.streetName;
+        searchService.doSearch = true;
+        $location.url('/');
+    };
+
+    uiGmapGoogleMapApi.then(function (maps) {
+
+    });
+
+    $scope.putMarker = function (mapModel, location) {
+        if (!$scope.marker) {
+            $scope.marker = new google.maps.Marker({
+                position: location,
+                map: mapModel
+            });
+        } else {
+            $scope.marker.setPosition(location);
+        }
+    }
+
+    $scope.map = {
+        // should use the one from the device, just a mock
+        center: {
+            latitude: -27.5964796,
+            longitude: -48.5205031
+        },
+        events: {
+            click: function (mapModel, eventName, originalEventArgs) {
+                var e = originalEventArgs[0];
+                $scope.geocode(e.latLng);
+                $scope.putMarker(mapModel, e.latLng);
+            },
+        },
+        zoom: 17
+    };
+
+    var geocoder = new google.maps.Geocoder();
+
+    $scope.geocode = function (latlng) {
+        geocoder.geocode({
+            'latLng': latlng
+        }, function (results, status) {
+            if (status == google.maps.GeocoderStatus.OK) {
+                if (results[1]) {
+                    $scope.streetName = results[0].address_components[1].long_name;
+                    $scope.$evalAsync();
+                } else {
+                    console.log('Location not found');
+                }
+            } else {
+                console.log('Geocoder failed due to: ' + status);
+            }
+        });
+    }
+
+}]);
+
+
+floripaApp.controller('SearchCtrl', ['$scope', 'searchService', '$location', function ($scope, searchService, $location) {
+
+    $scope.searchService = searchService;
```

Comment:
I am using it to bind values in the html part. This is not a regular service, I am using it for its singleton nature.